analyze/analyze.rs

```diff
@@ -939,18 +939,20 @@ impl traits::Emit for Monos {
                 bloat,
             } = entry;
 
+            let get_size_percent = move |x: u32| f64::from(x) / total_size * 100.0;
+
             vec![TableRow {
                 bloat: Some(*bloat),
-                bloat_percent: Some(f64::from(*bloat) / total_size * 100.0),
+                bloat_percent: Some(get_size_percent(*bloat)),
                 size: *size,
-                size_percent: f64::from(*size) / total_size * 100.0,
+                size_percent: get_size_percent(*size),
                 name: name.to_string(),
             }].into_iter()
```

Comment:
The `vec![..]` will still allocate the heap elements, what we want to do is

```rust
use std::iter;

iter::once(TableRow {
    ...
}).chain(...)
```

instead of using `vec![]` here.

Reply:
Omg! Okay *that's* how to do this! I had been having trouble with plain arrays, I didn't know about `iter::once`. I'll take care of that today, thanks for the tip!